Desktop/analysis/analyses.cpp

```diff
@@ -47,6 +47,14 @@ Analyses::Analyses()
 	new KnownIssues(this);
 }
 
+void Analyses::destroyAllForms() 
+{ 
+	Log::log() << "Analyses::destroyAllForms()" << std::endl;
+	
+	//Destroy all existing forms *before* destroying the rest of QML, to avoid a massive slew of errors
+	applyToAll([](Analysis * a){ if(a->form()) a->destroyForm(); });
```

Comment:
I looked a bit around, and I think you might be able to improve the implementation of these `Apply*` methods by using [`std::for_each`](https://en.cppreference.com/w/cpp/algorithm/for_each) or `std::transform`; and if that's doable, then you might be able to get some free parallelism by adding an `ExecutionPolicy` to it. 

I think it might be worth looking into, especially since this is a frequent action.

Reply:
Could be nice but really quite outside of the scope of this pr